libraries/DataFlash/LogFile.cpp

```diff
@@ -2086,3 +2086,16 @@ void DataFlash_Class::Log_Write_Rally(const AP_Rally &rally)
         }
     }
 }
+
+// Write AOA and SSA
```

Comment:
sorry, I should have explained better.
The method you have used produces minimal overhead logging, at the expense of more complex code, where you have to add bits of code in 3 places. We have a simpler mechanism where you can log something with just one line of code, like this:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_TECS/AP_TECS.cpp#L1063
it does cost a few extra microseconds as it has to construct the structure on the stack, but it makes for easier to read code